charts/opentelemetry-demo/templates/_pod.tpl

```diff
@@ -3,12 +3,12 @@ Get Pod Env
 Note: Consider that dependent variables need to be declared before the referenced env variable.
 */}}
 {{- define "otel-demo.pod.env" -}}
-{{- if .useDefault.env  }}
-{{ include "otel-demo.envOverriden" (dict "env" .defaultValues.env "envOverrides" .defaultValues.envOverrides "Template" $.Template) }}
-{{- end }}
 {{- if .env }}
 {{ include "otel-demo.envOverriden" . }}
 {{- end }}
+{{- if .useDefault.env  }}
```

Comment:
I think no matter which order we chose someone is gonna be unhappy.  I think wanting to add content to `OTEL_RESOURCE_ATTRIBUTES` is a good reason to enact this switch.

Also, the number of default env vars is small, so if someone needed to reference `OTEL_SERVICE_NAME` in a component it wouldn't be too hard to remove it from default and add it to the component's env vars.

Reply:
I think the proper foolproof solution is because OTEL_RESOURCE_ATTRIBUTES depends on Kubernetes env var expansion, we should find a way to force it to always be last.

Reply:
We could definitely do that by pulling it out of the default list, applying the default list, applying the component env vars, and then applying the `OTEL_RESOURCE_ATTRIBUTES` env var.  Do you want to try that instead of switching the order?

Reply:
rewrote the logic for this.

Does defaults env vars before component ones, but now it will pull OTEL_RESOURCE_ATTRIBUTES out of the default ones, and add it back in at the of the concatenated list.